src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PrivateDataRetrievalService.java

```diff
@@ -36,6 +36,10 @@ public void initialisePrivateData(String companyNumber)
     this.managingOfficerData = getManagingOfficerData(companyNumber);
   }
 
+  public void initialisePrivateOverseasEntityData(String companyNumber) throws ServiceException {
```

Comment:
It's a bug to remember the overseasEntityData. Making a method public will not disrupt anything.

Reply:
just go get it when you need it.

Reply:
I have to say I really disagree with this suggestion, it doesn't make any sense.

If there's a pattern there the pattern should be followed - it's why it's there in the first place, right? Just because a bug has been raised separately for an issue with that pattern, doesn't mean existing WIPs should all address it resulting in unnecessary scope creep.

Only if the bug fix goes in first, and the pattern then changes, should it fall under scope of this ticket to follow that new pattern. And vice versa.

Reply:
Okay so you guys are really talking about race conditions here? So looking at this... If two requests come in for two different companies, Company 1 first and Company 2 second, and Company 1 hasn't finished its processing when Company 2 comes in right behind it and initialises overseasEntityData... meaning Company 1 could now gets the data of Company 2 (in theory).

If I've understood that correctly then I would just have initialisePrivateOverseasEntityData() return it directly:
return getOverseasEntityData(companyNumber);
Because I spoke last I get the final say NaNaNa-Na-Na-Naaa

Reply:
Rebased off main which has Andre's fix for the race condition, so there are no changes here anymore :tada: